Can't use a global robot account to maintain harbor_config_security

**Describe the bug**
module.harbor_administration.harbor_config_security.main[0]: Modifying... [id=/system/CVEAllowlist]
╷
│ Error: [ERROR] no X-Harbor-Csrf-Token present to retry
│ 
│   with module.harbor_administration.harbor_config_security.main[0],
│   on modules/harbor-administration/main.tf line 18, in resource "harbor_config_security" "main":
│   18: resource "harbor_config_security" "main" {

**To Reproduce**
Generate a harbor with all global system + project privileges and try to apply a new CVE in the whitelist and an error is generated with X-Harbor-Csrf-Token .
If no update is made all is good.

If I try with my UI account which is superadmin all seems good.

**Expected behavior**
Maintaining CVE resources should be possible via a global robot account.

**Additional context**
Add any other context about the problem here.

- Provider Version v3.10.21
- Terraform Version v1.9.7 on darwin_arm64
- Harbor Version v2.11.0-70255684

Moin,

I'm seeing the same thing for a `harbor_project`. 

Provider Version: 3.10.23
Terraform Version: 1.8.5
Harbor Version: Version v2.13.2-436918f8

I have a very simple configuration, which only consists of:

```terraform
variable "docker_registry_username" {}
variable "docker_registry_password" {}

resource "harbor_registry" "old_docker_registry" {
  provider_name = "docker-registry"

  name        = "not relevant"
  endpoint_url = "some-value"

  access_id     = var.docker_registry_username
  access_secret = var.docker_registry_password
}

import {
  to = harbor_registry.old_docker_registry
  id = "/registries/1"
}

resource "harbor_project" "old_docker_registry" {
  name = "old-docker-registry"

  registry_id = harbor_registry.old_docker_registry.id
}

import {
  to = harbor_project.old_docker_registry
  id = "/projects/4"
}
```

If I remove the `harbor_project`, I can plan + apply just fine:

```log
harbor_registry.old_docker_registry: Importing... [id=/registries/1]
harbor_registry.old_docker_registry: Import complete [id=/registries/1]
harbor_registry.old_docker_registry: Modifying... [id=/registries/1]
harbor_registry.old_docker_registry: Modifications complete after 0s [id=/registries/1]

Apply complete! Resources: 1 imported, 0 added, 1 changed, 0 destroyed.
```

If I add the `harbor_project`, a `terraform plan` fails during the plan:

```log
harbor_registry.old_docker_registry: Refreshing state... [id=/registries/1]
harbor_project.old_docker_registry: Preparing import... [id=/projects/4]
harbor_project.old_docker_registry: Refreshing state... [id=/projects/4]

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: [ERROR] no X-Harbor-Csrf-Token present to retry
│ 
```

--- a/client/client.go
+++ b/client/client.go
@@ -103,18 +103,21 @@ func (c *Client) SendRequest(method string, path string, payload interface{}, st
 	}
 
 	if resp.StatusCode == http.StatusForbidden {
-		csrfheaders, err := extractCsrfHeaders(resp.Header)
-		if err != nil {
-			return "", "", resp.StatusCode, err
-		}
-		resp, err = c.sendRequestWithHeaders(method, path, payload, csrfheaders)
-		if err != nil {
-			if resp != nil {
+		if c.sessionId != "" {
+			csrfheaders, err := extractCsrfHeaders(resp.Header)
+			if err != nil {
 				return "", "", resp.StatusCode, err
-			} else {
-				return "", "", http.StatusBadGateway, err
+			}
+			resp, err = c.sendRequestWithHeaders(method, path, payload, csrfheaders)
+			if err != nil {
+				if resp != nil {
+					return "", "", resp.StatusCode, err
+				} else {
+					return "", "", http.StatusBadGateway, err
+				}
 			}
 		}
+		return "", "", resp.StatusCode, fmt.Errorf("[ERROR] forbidden: status=%s, code=%d \nIf you are using a robot account, this is likely due to RBAC limitations. See: https://github.com/goharbor/community/blob/main/proposals/new/Robot-Account-Expand.md", resp.Status, resp.StatusCode)
 	}
 
 	body, err := io.ReadAll(resp.Body)

--- a/provider/resource_config_auth.go
+++ b/provider/resource_config_auth.go
@@ -197,7 +197,9 @@ func resourceConfigAuthRead(d *schema.ResourceData, m interface{}) error {
 	resp, _, respCode, err := apiClient.SendRequest("GET", models.PathConfig, nil, 200)
 	if respCode == 404 && err != nil {
 		d.SetId("")
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
+	} else if err != nil {
+		return err
 	}
 
 	err = client.SetAuthValues(d, resp)

--- a/provider/resource_config_security.go
+++ b/provider/resource_config_security.go
@@ -61,6 +61,8 @@ func resourceConfigSecurityRead(d *schema.ResourceData, m interface{}) error {
 	if respCode == 404 && err != nil {
 		d.SetId("")
 		return fmt.Errorf("resource not found %s", models.PathSystemCVEAllowList)
+	} else if err != nil {
+		return err
 	}
 
 	var jsonData models.SystemCveAllowListBodyPost

--- a/provider/resource_config_system.go
+++ b/provider/resource_config_system.go
@@ -131,6 +131,8 @@ func resourceConfigSystemRead(d *schema.ResourceData, m interface{}) error {
 	if respCode == 404 && err != nil {
 		d.SetId("")
 		return fmt.Errorf("error getting system configuration %s", err)
+	} else if err != nil {
+		return err
 	}
 
 	var jsonData models.ConfigBodyResponse

--- a/provider/resource_garbage_collection.go
+++ b/provider/resource_garbage_collection.go
@@ -57,7 +57,9 @@ func resourceGCRead(d *schema.ResourceData, m interface{}) error {
 	resp, _, respCode, err := apiClient.SendRequest("GET", models.PathGC, nil, 200)
 	if respCode == 404 && err != nil {
 		d.SetId("")
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
+	} else if err != nil {
+		return err
 	}
 
 	var jsonData models.SystemBody

--- a/provider/resource_group.go
+++ b/provider/resource_group.go
@@ -62,13 +62,13 @@ func resourceGroupRead(d *schema.ResourceData, m interface{}) error {
 		d.SetId("")
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("resource not found %s", d.Id())
+		return err
 	}
 
 	var jsonData models.GroupBody
 	err = json.Unmarshal([]byte(resp), &jsonData)
 	if err != nil {
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
 	}
 
 	d.Set("group_name", jsonData.Groupname)

--- a/provider/resource_harbor_project_webhook.go
+++ b/provider/resource_harbor_project_webhook.go
@@ -77,7 +77,7 @@ func resourceProjectWebhookCreate(d *schema.ResourceData, m interface{}) error {
 		return err
 	}
 
-	id, err := client.GetID(headers)
+	id, _ := client.GetID(headers)
 	d.SetId(id)
 	return resourceProjectWebhookRead(d, m)
 }
@@ -88,13 +88,15 @@ func resourceProjectWebhookRead(d *schema.ResourceData, m interface{}) error {
 	resp, _, respCode, err := apiClient.SendRequest("GET", d.Id(), nil, 200)
 	if respCode == 404 && err != nil {
 		d.SetId("")
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
+	} else if err != nil {
+		return err
 	}
 
 	var jsonData models.ProjectWebhook
 	err = json.Unmarshal([]byte(resp), &jsonData)
 	if err != nil {
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
 	}
 
 	d.Set("name", jsonData.Name)

--- a/provider/resource_interrogation_services.go
+++ b/provider/resource_interrogation_services.go
@@ -51,13 +51,15 @@ func resourceVulnRead(d *schema.ResourceData, m interface{}) error {
 
 	if respCode == 404 && err != nil {
 		d.SetId("")
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
+	} else if err != nil {
+		return err
 	}
 
 	var jsonData models.InterogationsBodyResponse
 	err = json.Unmarshal([]byte(resp), &jsonData)
 	if err != nil {
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
 	}
 
 	vulnerability_scan_policy := jsonData.Schedule.Type

--- a/provider/resource_labels.go
+++ b/provider/resource_labels.go
@@ -53,7 +53,7 @@ func resourceLabelCreate(d *schema.ResourceData, m interface{}) error {
 		return err
 	}
 
-	id, err := client.GetID(headers)
+	id, _ := client.GetID(headers)
 	d.SetId(id)
 	return resourceLabelRead(d, m)
 }
@@ -66,13 +66,13 @@ func resourceLabelRead(d *schema.ResourceData, m interface{}) error {
 		d.SetId("")
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("resource not found %s", d.Id())
+		return err
 	}
 
 	var jsonData models.Labels
 	err = json.Unmarshal([]byte(resp), &jsonData)
 	if err != nil {
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
 	}
 
 	d.Set("name", jsonData.Name)

--- a/provider/resource_log_rotation.go
+++ b/provider/resource_log_rotation.go
@@ -51,7 +51,10 @@ func resourcePurgeAuditRead(d *schema.ResourceData, m interface{}) error {
 	if respCode == 404 && err != nil {
 		d.SetId("")
 		return fmt.Errorf("resource not found %s", d.Id())
+	} else if err != nil {
+		return err
 	}
+
 	if len(resp) == 0 {
 		d.SetId("")
 		return nil
@@ -103,7 +106,7 @@ func validateIncludeOperations(v interface{}, k string) (warns []string, errs []
 	for _, op := range ops {
 		op = strings.TrimSpace(op)
 		if !containsString(validValues, op) {
-			errs = append(errs, fmt.Errorf("Invalid value %q in %q. Valid values are: create, pull, delete", op, k))
+			errs = append(errs, fmt.Errorf("invalid value %q in %q. Valid values are: create, pull, delete", op, k))
 		}
 	}
 

--- a/provider/resource_preheat__instance.go
+++ b/provider/resource_preheat__instance.go
@@ -101,7 +101,7 @@ func resourcePreheatInstanceCreate(d *schema.ResourceData, m interface{}) error 
 		return err
 	}
 
-	id, err := client.GetID(headers)
+	id, _ := client.GetID(headers)
 	d.SetId(id)
 	return resourcePreheatInstanceRead(d, m)
 }
@@ -114,13 +114,13 @@ func resourcePreheatInstanceRead(d *schema.ResourceData, m interface{}) error {
 		d.SetId("")
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("resource not found %s", d.Id())
+		return err
 	}
 
 	var jsonData models.PreheatInstance
 	err = json.Unmarshal([]byte(resp), &jsonData)
 	if err != nil {
-		return fmt.Errorf("Resource not found %s", d.Id())
+		return fmt.Errorf("resource not found %s", d.Id())
 	}
 
 	d.Set("name", jsonData.Name)

--- a/provider/resource_project.go
+++ b/provider/resource_project.go
@@ -237,6 +237,8 @@ func resourceProjectDelete(d *schema.ResourceData, m interface{}) error {
 	_, _, respCode, err := apiClient.SendRequest("DELETE", d.Id(), nil, 200)
 	if respCode != 404 && err != nil { // We can't delete something that doesn't exist. Hence the 404-check
 		return err
+	} else if err != nil {
+		return err
 	}
 	return nil
 }

--- a/provider/resource_project_member_group.go
+++ b/provider/resource_project_member_group.go
@@ -103,7 +103,7 @@ func resourceMembersGroupRead(d *schema.ResourceData, m interface{}) error {
 		d.SetId("")
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("resource not found %s", d.Id())
+		return err
 	}
 
 	var jsonData models.ProjectMembersBodyResponses

--- a/provider/resource_project_member_user.go
+++ b/provider/resource_project_member_user.go
@@ -82,7 +82,7 @@ func resourceMembersUserRead(d *schema.ResourceData, m interface{}) error {
 		d.SetId("")
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("resource not found %s", d.Id())
+		return err
 	}
 
 	var jsonData models.ProjectMembersBodyResponses

--- a/provider/resource_registry.go
+++ b/provider/resource_registry.go
@@ -92,7 +92,7 @@ func resourceRegistryRead(d *schema.ResourceData, m interface{}) error {
 		d.SetId("")
 		return nil
 	} else if err != nil {
-		return fmt.Errorf("resource not found %s", d.Id())
+		return err
 	}
 
 	var jsonData models.RegistryBody
@@ -133,6 +133,8 @@ func resourceRegistryDelete(d *schema.ResourceData, m interface{}) error {
 
 	_, _, respCode, err := apiClient.SendRequest("DELETE", d.Id(), nil, 200)
 	if respCode != 404 && err != nil { // We can't delete something that doesn't exist. Hence the 404-check
+		return err
+	} else if err != nil {
 		return err
 	}
 	return nil

--- a/provider/resource_replication.go
+++ b/provider/resource_replication.go
@@ -229,7 +229,6 @@ func resourceReplicationUpdate(d *schema.ResourceData, m interface{}) error {
 		}
 
 		_, _, _, err := apiClient.SendRequest("POST", models.PathExecution, body, 201)
-
 		if err != nil {
 			return err
 		}
@@ -243,6 +242,8 @@ func resourceReplicationDelete(d *schema.ResourceData, m interface{}) error {
 
 	_, _, respCode, err := apiClient.SendRequest("DELETE", d.Id(), nil, 200)
 	if respCode != 404 && err != nil { // We can't delete something that doesn't exist. Hence the 404-check
+		return err
+	} else if err != nil {
 		return err
 	}
 	return nil

--- a/provider/resource_robot_account.go
+++ b/provider/resource_robot_account.go
@@ -175,7 +175,10 @@ func resourceRobotAccountRead(d *schema.ResourceData, m interface{}) error {
 			if respCode == 404 && err != nil {
 				d.SetId("")
 				return fmt.Errorf("error getting system configuration (probably missing admin rights) %s, you can use robot_prefix to force the prefix", err)
+			} else if err != nil {
+				return err
 			}
+
 			var systemConfig models.ConfigBodyResponse
 			err = json.Unmarshal([]byte(resp), &systemConfig)
 			if err != nil {

--- a/provider/resource_tasks.go
+++ b/provider/resource_tasks.go
@@ -102,8 +102,10 @@ func resourceTasksUpdate(d *schema.ResourceData, m interface{}) error {
 		},
 	}
 
-	apiClient.SendRequest("PUT", pathVuln, body, 0)
-
+	_, _, _, err := apiClient.SendRequest("PUT", pathVuln, body, 0)
+	if err != nil {
+		return err
+	}
 	return resourceTasksRead(d, m)
 }
 
@@ -114,7 +116,10 @@ func resourceTasksDelete(d *schema.ResourceData, m interface{}) error {
 		Schedule: cron{Cron: ""},
 	}
 
-	apiClient.SendRequest("PUT", pathVuln, body, 0)
+	_, _, _, err := apiClient.SendRequest("PUT", pathVuln, body, 0)
+	if err != nil {
+		return err
+	}
 	return nil
 }
 

